game-loop/src/main/java/com/iluwatar/gameloop/BasicGamePanel.java

```diff
@@ -0,0 +1,106 @@
+/**
+ * The MIT License
+ * Copyright (c) 2014 Ilkka Seppälä
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package com.iluwatar.gameloop;
+
+import javax.swing.*;
+import java.awt.*;
+
+/**
+ * The basic game panel used for most of the game loops shown in this package.
+ * This class includes the game logic as well as the paint instructions
+ * used to render the animation.
+ */
+public class BasicGamePanel extends JPanel {
+
+  int ballX;
+  int ballY;
+  int lastBallX;
+  int lastBallY;
+  int ballWidth;
+  int ballHeight;
+  private float ballXVel;
+  private float ballYVel;
+  private double bounceDepreciation = 1;
+
+  int lastDrawX;
+  int lastDrawY;
+
+  public void setInterpolation(float interpolation) {}
```

Comment:
This points to a design problem, that even though panel is not an interpolated one we have empty setter. The solution to that is to use generic type in `BallAnimation`.
```
public class BallAnimation<T extends BasicGamePanel> extends JFrame {
   T gPanel; // Use generic type so that if extending class gets the panel it will get the type of panel declared in class definition
   
    public BallAnimation(T gPanel) {
       this.gPanel = gPanel;
    }
}

public class IndependentAnimation extends BallAnimation<InterpolatedPanel> {
   
   void updateView(float interpolation) {
       gPanel.setInterpolation(interpolation); // Here we get InterpolatedPanel, so setInterpolation method only needs to be present in InterpolatedPanel class
   }
}
```
Make respective changes of in other Animation classes regarding raw type.